R/epiparameter_db.R

```diff
@@ -0,0 +1,765 @@
+  }
+
+  # subset by subset conditions
+  if (is.call(expr)) {
+    nse_subject <- as.character(expr)[2]
+    cond_list <- lapply(
+      multi_epiparameter,
+      .is_cond_epiparameter,
+      expr,
+      nse_subject
+    )
+    set <- vapply(cond_list, function(x) any(unlist(x)), FUN.VALUE = logical(1))
+    set[is.na(set)] <- FALSE
+    multi_epiparameter <- multi_epiparameter[set]
+  } else if (is.function(subset)) {
+    set <- vapply(multi_epiparameter, subset, FUN.VALUE = logical(1))
+    multi_epiparameter <- multi_epiparameter[set]
+  }
+
+  attributes(multi_epiparameter) <- attrib
+
+  if (length(multi_epiparameter) == 0) {
+    stop(
+      "No entries in the database meet the search criteria.",
+      call. = FALSE
+    )
+  }
+
+  lapply(multi_epiparameter, assert_epiparameter)
+  is_param <- vapply(
+    multi_epiparameter,
+    is_parameterised,
+    FUN.VALUE = logical(1)
+  )
+
+  if (single_epiparameter) {
+    # select parameterised entries
+    if (sum(is_param) >= 1) {
+      multi_epiparameter <- multi_epiparameter[is_param]
+    }
+    # select entries that accounted for truncation
+    idx <- vapply(
+      multi_epiparameter,
+      function(x) x$method_assess$right_truncated,
+      FUN.VALUE = logical(1)
+    )
+    if (sum(idx) >= 1) {
+      multi_epiparameter <- multi_epiparameter[idx]
+    }
+    # select largest sample size
+    idx <- which.max(
+      vapply(
+        multi_epiparameter,
+        function(x) x$metadata$sample_size,
+        FUN.VALUE = numeric(1)
+      )
+    )
+    single_epiparameter <- multi_epiparameter[[idx]]
+
+    message(
+      "Using ", format(get_citation(single_epiparameter)), ". \n",
+      "To retrieve the citation use the 'get_citation' function"
+    )
+
+    return(single_epiparameter)
+  }
+
+  message(
+    "Returning ", length(multi_epiparameter), " results that match the ",
+    "criteria (", sum(is_param), " are parameterised). \n",
+    "Use subset to filter by entry variables or ",
+    "single_epiparameter to return a single entry. \n",
+    "To retrieve the citation for each use the ",
+    "'get_citation' function"
+  )
+
+  if (length(multi_epiparameter) == 1) {
+    multi_epiparameter <- multi_epiparameter[[1]]
+  }
+
+  # return epiparameter
+  multi_epiparameter
+}
+
+#' @inherit epiparameter_db title
+#'
+#' @description
+#' `r lifecycle::badge('deprecated')`
+#'
+#' `epidist_db()` has been renamed [epiparameter_db()]. Please use
+#' `epiparameter_db()` instead as the `epidist_db()` alias will be removed from
+#' the package in the future.
+#'
+#' @inheritParams epiparameter_db
+#'
+#' @inherit epiparameter_db return
+#' @export
+#' @keywords internal
+epidist_db <- function(disease = "all",
+                       pathogen = "all",
+                       epi_name = "all",
+                       author = NULL,
+                       subset = NULL,
+                       single_epiparameter = FALSE) {
+  lifecycle::deprecate_warn(
```

Comment:
What version do you want to remove this by? Helpful to note that the deprecation will be completed by a certain version so that you don't get stuck keeping a deprecated function for a long time. Also helps people know by when they need to have moved on.

Good practice on deprecating 👍

Reply:
I'll remove this function in v0.5.0. I've added a details note to `lifecycle::deprecate_warn()` in PR #399 to tell users about when the function will be removed. In {epiparameter} v0.4.0 I will change `lifecycle::deprecate_warn()` to `lifecycle::deprecate_stop()`.